packages/react/src/components/CopyButton/CopyButton-test.js

```diff
@@ -5,97 +5,112 @@
-      const feedback = feedbackWrapper.find(`.${prefix}--copy-btn__feedback`);
-      expect(feedback).toEqual({});
-      feedbackWrapper.simulate('click');
-      expect(feedback).toBeTruthy();
-    });
-
-    it('Should show feedback for a limited amount of time', () => {
-      const feedbackWrapper = mount(
-        <CopyButton feedback="Copied!" feedbackTimeout={5000} />
-      );
-      feedbackWrapper.simulate('click');
-      const copyButton = feedbackWrapper.find('button');
-      expect(copyButton.hasClass(`${prefix}--copy-btn--animating`)).toBe(true);
-      setTimeout(() => {
-        expect(copyButton.hasClass(`${prefix}--copy-btn--animating`)).toBe(
-          false
-        );
-      }, 5220); // 5000 + 2 * 110 (transition duration)
-    });
+  it('should call the click handler', () => {
+    const onClick = jest.fn();
+
+    render(
+      <CopyButton
+        iconDescription="icon description"
+        data-testid="copy-btn-4"
+        onClick={onClick}
+      />
+    );
+    const button = screen.getByTestId('copy-btn-4');
+    userEvent.click(button);
+    expect(onClick).toHaveBeenCalled();
   });
+});
+
+describe('Feedback', () => {
+  it('should make the feedback visible for a limited amount of time', () => {
+    render(
+      <CopyButton iconDescription="icon description" data-testid="copy-btn-5" />
+    );
+
+    const button = screen.getByTestId('copy-btn-5');
+    userEvent.click(button);
+
+    expect(button).toHaveClass('cds--copy-btn--animating');
+    setTimeout(() => {
+      expect(button).not.toHaveClass('cds--copy-btn--animating');
+    }, 2000); // default feedback timeout
+  });
+
+  it('should be able to specify the feedback message', () => {
+    render(
+      <CopyButton
+        iconDescription="icon description"
+        data-testid="copy-btn-6"
+        feedback="custom-feedback"
+      />
+    );
+
+    const button = screen.getByTestId('copy-btn-6');
+    userEvent.click(button);
+    setTimeout(() => {
+      expect(screen.getByLabelText('custom-feedback')).toBeInTheDocument();
+    }, 1000); //testing during the feedback window
+  });
+
+  it('should allow users to override default feedback timeout via prop', () => {
+    render(
+      <CopyButton
+        iconDescription="icon description"
+        data-testid="copy-btn-7"
+        feedbackTimeout={5000}
+      />
+    );
+
+    const button = screen.getByTestId('copy-btn-7');
+    userEvent.click(button);
 
-  describe('Triggers appropriate events', () => {
-    it('should call the click handler', () => {
-      const onClick = jest.fn();
-      const clickWrapper = mount(<CopyButton onClick={onClick} />);
-      clickWrapper.simulate('click');
-      expect(onClick).toHaveBeenCalled();
-    });
+    expect(button).toHaveClass('cds--copy-btn--animating');
+    setTimeout(() => {
+      expect(button).not.toHaveClass('cds--copy-btn--animating');
+    }, 5000); // custom feedback timeout
```

Comment:
For timers, could you update this to use [timer mocks](https://jestjs.io/docs/timer-mocks) (just so we don't have to wait 5s for the test to complete)